File: Task/Jewels-and-stones/Visual-Basic-.NET/jewels-and-stones.vb
Module Module1

    Function Count(stones As String, jewels As String) As Integer
        Dim bag = jewels.ToHashSet
        Return stones.Count(AddressOf bag.Contains)
    End Function

    Sub Main()
        Console.WriteLine(Count("aAAbbbb", "Aa"))
        Console.WriteLine(Count("ZZ", "z"))
    End Sub

End Module

File: Task/Jewels-and-stones/C-sharp/jewels-and-stones.cs
using System;
using System.Linq;

public class Program
{
    public static void Main() {
        Console.WriteLine(Count("aAAbbbb", "Aa"));
        Console.WriteLine(Count("ZZ", "z"));
    }

    private static int Count(string stones, string jewels) {
        var bag = jewels.ToHashSet();
        return stones.Count(bag.Contains);
    }
}
